build/common.go

```diff
@@ -50,26 +60,73 @@ func getExecutableName(os string, arch string, pluginJSONPath string) (string, e
 		exname = exename
 	}
 
-	exeName := fmt.Sprintf("%s_%s_%s", exname, os, arch)
+	return asExecutableName(os, arch, pluginJSONPath), nil
+}
+
+// execNameCache is a cache for the executable name, so we don't have to read the plugin.json file multiple times.
+var executableNameCache sync.Map
```

Comment:
This is mimicking what we are already doing today with `var exname string` above

Reply:
I guess for building for all distros for example you benefit from not reading it 6+ times. I haven't performance tested it or anything though 😅

Reply:
I did some light testing and it's negligible it seems. I'll keep it for now for consistency and it does not harm I guess